source/grid/grid_tools_dof_handlers.cc

```diff
@@ -2142,29 +2143,29 @@ namespace GridTools
     unsigned int n_matches = 0;
 
     // Match with a complexity of O(n^2). This could be improved...
-    std::bitset<3> orientation;
     using PairIterator =
       typename std::set<std::pair<CellIterator, unsigned int>>::const_iterator;
     for (PairIterator it1 = pairs1.begin(); it1 != pairs1.end(); ++it1)
       {
         for (PairIterator it2 = pairs2.begin(); it2 != pairs2.end(); ++it2)
           {
-            const CellIterator cell1     = it1->first;
-            const CellIterator cell2     = it2->first;
-            const unsigned int face_idx1 = it1->second;
-            const unsigned int face_idx2 = it2->second;
-            if (GridTools::orthogonal_equality(orientation,
-                                               cell1->face(face_idx1),
-                                               cell2->face(face_idx2),
-                                               direction,
-                                               offset,
-                                               matrix))
+            const CellIterator                  cell1     = it1->first;
+            const CellIterator                  cell2     = it2->first;
+            const unsigned int                  face_idx1 = it1->second;
+            const unsigned int                  face_idx2 = it2->second;
+            const std::optional<std::bitset<3>> orientation =
+              GridTools::orthogonal_equality(cell1->face(face_idx1),
+                                             cell2->face(face_idx2),
+                                             direction,
+                                             offset,
+                                             matrix);
+            if (orientation)
```

Comment:
You can now write this as follows:
```suggestion
            if (const std::optional<std::bitset<3>> orientation =
                GridTools::orthogonal_equality(cell1->face(face_idx1),
                                             cell2->face(face_idx2),
                                             direction,
                                             offset,
                                             matrix);
              orientation.has_value())
```

Reply:
I think those two lines do the same thing: I opted to do the assignment in the if statement (like what we do with `dynamic_cast`).